source/Hash.h

```diff
@@ -33,6 +33,9 @@
 
 #ifndef HASH_INCLUDED
 #define HASH_INCLUDED
+
+#if __cplusplus < 201103L
```

Comment:
Are there other c++11 versions besides March (03)? If not, I'd add a comment like "// c++11" or something to clarify. (I've never used these values, but I'd not heard it called 201103 before).